### PR TITLE
[PWA-1029] HomePage unstyled in scaffolded app

### PR DESCRIPTION
**Pull request by mikhaelbois**
## Description

By default, when building an app, the edition of the Magento Store is set to `CE`.
The custom CSS for the Home Page is only called in the `EE` version.

With this update, the default will now be `EE` and we will also ask the question when creating the app.

## Related Issue

Closes https://jira.corp.magento.com/browse/PWA-1029.

#### Test scenario(s) for direct fix/feature

- Test scaffolding. Remove or edit the `--backend-edition` parameter. The .env file should reflect your changes.
- Run `packages/pwa-buildpack/bin/buildpack create-project scaffold-test --template "@magento/venia-concept" --name "scaffold-test" --author "test123 <tester@example.com>" --backend-url "https://master-7rqtwti-c5v7sxvquxwl4.eu-4.magentosite.cloud/" --braintree-token "sandbox_8yrzsvtm_s2bg8fs563crhqzk" --npm-client "yarn" --install false --backend-edition="EE"`.
- Test project creation. You should have a prompt asking you for the Magento Edition. The .env file should reflect your choices.
- Run `packages/create-pwa/bin/create-pwa`.

#### Test scenario(s) for any existing impacted features/areas

- Run `yarn test`.
- Run `cd packages/create-pwa/ && yarn jest`.

### CODE REVIEW

File: packages/pwa-buildpack/lib/__tests__/cli-create-project.spec.js [hunk 1]
```diff
@@ -213,7 +213,107 @@ test('warns if backendurl does not match env', async () => {
...
+});
+
+test('if backendEdition matches env', async () => {
+    const old = process.env.MAGENTO_BACKEND_EDITION;
+    process.env.MAGENTO_BACKEND_EDITION = 'EE';
+    fse.ensureDir.mockResolvedValueOnce(true);
+    fse.readdir.mockResolvedValueOnce(true);
+    await expect(
+        createProjectCliBuilder.handler({
+            backendUrl: 'EE',
```
Comment (JustinConabreeAbsolunet) [inline packages/pwa-buildpack/lib/__tests__/cli-create-project.spec.js:263]: Should backendUrl be `https://example.com` instead?

Comment (mikhaelbois) [reply packages/pwa-buildpack/lib/__tests__/cli-create-project.spec.js:263]: True, wrong copy paste. I'll fix that.

Comment (mikhaelbois) [reply packages/pwa-buildpack/lib/__tests__/cli-create-project.spec.js]: @JustinConabreeAbsolunet Fixed.

**Comment by revanth0212** [comment]
Nice work @mikhaelbois. I haven't reviewed the code yet, but I would like to leave a note here that we need to update our CI jobs and danger CI code to implement answers for the new question added into this PR. If not the jobs will fail.

**Comment by revanth0212** [comment]
Working 👏 

Command: DEBUG_PROJECT_CREATION=1 node ../pwa-studio/packages/create-pwa/bin/create-pwa 

image

Result: 
image

One thing I am concerned about is, not sure why but `npm` was selected as the default package manager, it should be `yarn` instead. We recently changed that in a PR #3363

**Review by revanth0212 — CHANGES_REQUESTED** [review]
I am cool with the changes, they look straight forward and thanks for covering your changes and the changes that were missed before. Please look into why `npm` was selected by default as the package manager instead of `yarn`, its kind of a big deal for the CI to work. Other than that the PR is ready for QA.

**Review by revanth0212 — APPROVED** [review]
My bad. Your PR is fine. 

> Please look into why npm was selected by default as the package manager instead of yarn, its kind of a big deal for the CI to work. Other than that the PR is ready for QA.

It's because we have a check to look at the user agent and in this case, I used `node` to run the script and that is the reason why it chose `npm` instead. Ill fix this in a different PR.

**Comment by revanth0212** [comment]
Created https://github.com/magento-commerce/pwa-studio-cicd/pull/84 on the CICD repo to cover the scaffolding changes introduced in this PR. We need to merge this PR as well as part of this ticket.

File: packages/pwa-buildpack/lib/__tests__/cli-create-project.spec.js [hunk 2]
```diff
@@ -213,7 +213,107 @@ test('warns if backendurl does not match env', async () => {
     process.env.MAGENTO_BACKEND_URL = old;
 });
 
-test('runs install', async () => {
+test('if backendUrl matches env', async () => {
```
[the remark this answers is not in the document]
Comment (mikhaelbois) [reply packages/pwa-buildpack/lib/__tests__/cli-create-project.spec.js]: I've updated the tests description to tell that it does not show any warnings if it matches env.

**Comment by tjwiebell** [comment]
This is QA approved ✅  just need re-approval from @revanth0212 and confirmation magento-commerce/pwa-studio-cicd#84 is approved and ready to merge as well.

**Review by revanth0212 — APPROVED** [review]

**Pull request merged**